lib/TestReportHandler.rb

```diff
@@ -127,11 +127,23 @@ def pass
     return 
   end 
 
+#	Might need to alter ‘score’ calculation based on including/not including setup or teardown failures
+# Or, add the option to decide whether to include them
+
   def finalize
     report.status = 'completed'
     report.score = ((1 - @num_failures / @num_tests.to_f) * 100).round(2)
     report.result = (report.score == 100 ? 'pass' : 'fail')
     report.issued = DateTime.now
+    
+    num_passes = 0
+    report.test[0].action.each do |ac|
+      num_passes += 1 if ac.operation.result.code == 'pass' if ac.operation
+      num_passes += 1 if ac.assert.result.code == 'pass' if ac.assert
+    end
+
+    score_testonly = (num_passes / report.test[0].action.length.to_f * 100).round(2)
```

Comment:
Same thing here with just accessing the first Test element. I would just count the total number of action elements while in the loop you wrote above.

Reply:
Got it

Reply:
This still needs to be addressed. It's still just accessing the first element of `test`